src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,78 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const removeTodo = () => dispatch(currentTodoActions.removeTodo());
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }, [currentTodo]);
+
+  const handleClose = () => {
+    removeTodo();
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {loading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo?.id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleClose}
```

Comment:
```suggestion
              onClick={removeTodo}
```
`handleClose` function wrapper is redundant